src/plugincontainer.cpp

```diff
@@ -769,6 +781,193 @@ const PreviewGenerator &PluginContainer::previewGenerator() const
   return m_PreviewGenerator;
 }
 
+void PluginContainer::simulateStartup(const std::vector<IPlugin*>& plugins) const
+{
+  // setUserInterface()
+  if (m_UserInterface) {
+    for (auto* plugin : plugins) {
+      if (auto* proxy = dynamic_cast<IPluginProxy*>(plugin)) {
+        proxy->setParentWidget(m_UserInterface->mainWindow());
+      }
+      if (auto* modPage = dynamic_cast<IPluginModPage*>(plugin)) {
+        modPage->setParentWidget(m_UserInterface->mainWindow());
+      }
+      if (auto* tool = dynamic_cast<IPluginTool*>(plugin)) {
+        tool->setParentWidget(m_UserInterface->mainWindow());
+      }
+      if (auto* installer = dynamic_cast<IPluginInstaller*>(plugin)) {
+        installer->setParentWidget(m_UserInterface->mainWindow());
+      }
+    }
+  }
+
+  // Simulate callbacks, e.g. onUserInterfaceInitialized and onProfileChanged.
+  for (auto* plugin : plugins) {
+    auto* oproxy = organizerProxy(plugin);
+    oproxy->m_UserInterfaceInitialized(m_UserInterface->mainWindow());
+    oproxy->m_ProfileChanged(nullptr, m_Organizer->currentProfile());
+  }
+}
+
+void PluginContainer::loadPlugin(QString const& filepath)

```

Comment:
Again, this seems to be a repeat of `loadPlugins()`, can it be reused instead?

Reply:
Done.